### ISSUE TITLE
Keystone auth token used for openstack cloudprovider resources

### ISSUE THREAD

**Issue by chmouel** [kubernetes/kubernetes#14809]
I am looking at the keystone experimental auth :

https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/auth/authenticator/request/keystone/keystone.go#L36-L40

and at how OpenStack provider is implemented :

https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/openstack/openstack.go#L117-L130

The two doesn't seem to be connected which mean the user auth to your keystone but that auth is not reused for creating openstack resources (when that token could be straight used for it).

In other words being able to have the user authenticated via OpenStack Keystone in kube and have the OpenStack resources (i.e: cinder for the PV driver) used in Kube consumed from the authenticated user and not from a global user.

**Comment by liggitt** [comment]
That could be problematic... many of the interaction points for cloud providers aren't directly connected to a single user. For example, if one user creates a replication controller and another user changes its replica setting, which user's credentials would you expect to be used when creating volumes for the resulting pods?

**Comment by chmouel** [comment]
@liggitt  okay that make sense.

So for something else, what about having a project being able to specify a [delegated token ](http://techs.enovance.com/5858/role-delegation-in-keystone-trusts) and having the openstack cloud resources to that project used with that token ?

**Comment by liggitt** [comment]
s/project/namespace/ :)
I'm not as familiar with the cloud provider interfaces... not sure what would make sense

**Comment by ashw7n** [comment]
@uruddarraju

**Comment by ashw7n** [comment]
Currently cloud providers are configured statically, as you pointed out. But once we start supporting  dynamically acquired cloud resources (such as volumes) we might need to support passing the credentials through.

**Comment by xsgordon** [comment]
Despite its age I believe this is still a valid issue.

**Comment by dims** [comment]
Looks like one of the URL is stale, better url is https://blog.zhaw.ch/icclab/trust-delegation-in-openstack-using-keystone-trusts/

**Comment by dims** [comment]
@bugosiphionah Want to take a crack at this one? Need to come up with a design first. We can talk on sig openstack slack when you are free
